### ISSUE TITLE
 Redeploy Berkeley testnet

### ISSUE THREAD

**Issue by stevenplatt** [MinaProtocol/mina#12721]
This issue is to document the redeploymet of the Berkeley testnet. This redeployment was also shared in a Google Hangouts meeting. A full recording of the redeployment is available at the meeting link:

Berkeley Testnet Redeploy - Screenshare
Friday, February 24 · 11:00am – 3:00pm PST
Google Meet joining info
Video call link: https://meet.google.com/fsd-njac-iof

----------------

Deployment release notes:
```
Mina's ecosystem partner, O(1) Labs, is continuously working on improvements to the core protocol and today we are excited to announce an updated unstable release 2884275 that is ready to test on the Berkeley testnet! Please report any issues in the comments on this post or tag any issues you create with berkeley-rampup and include berkeley 2884275 in the issue title.

Improvements/features:
* Added a new field `implicit_account_creation_fee` to account updates, which can be used (on non-token parties) to subtract the creation fee from the transferred amount. https://github.com/MinaProtocol/mina/pull/10842
* Improved staking ledger sync during bootstrap https://github.com/MinaProtocol/mina/pull/11971
* Fix to prevent transaction pool DDOS https://github.com/MinaProtocol/mina/pull/12548
* Added an `access` permission to accounts https://github.com/MinaProtocol/mina/pull/12003
* Removed redundant fields `timestamp` and `global_slot_since_hard_fork` from zkApp preconditions https://github.com/MinaProtocol/mina/pull/12234
* Removed `transaction_expiry_hr` used for evicting long pending transaction from pool https://github.com/MinaProtocol/mina/pull/12231
* Added `valid_while` precondition to zkapp account updates https://github.com/MinaProtocol/mina/pull/12353
* Limit length of zkapp URIs to 255 characters https://github.com/MinaProtocol/mina/pull/12346
* Renames sequence events to actions https://github.com/MinaProtocol/mina/pull/12367
* Allow verification key update and usage in one zkapp transaction https://github.com/MinaProtocol/mina/pull/12398
* Add tokens from failed txns to archive DB https://github.com/MinaProtocol/mina/pull/12435
* Fix for https://github.com/MinaProtocol/mina/issues/12286 https://github.com/MinaProtocol/mina/pull/12450
* Ignore zkApp proofs for failed transactions during block validation https://github.com/MinaProtocol/mina/pull/12490
* Updated token call logic https://github.com/MinaProtocol/mina/pull/12512
* Specify verification key hash for proof updates https://github.com/MinaProtocol/mina/pull/12380
* New account permission to set vesting schedules https://github.com/MinaProtocol/mina/pull/12492
* Remove supercharged rewards https://github.com/MinaProtocol/mina/pull/12587
* Transaction hash prefix changed https://github.com/MinaProtocol/mina/pull/12651

Please refer to the schema from the released commit

https://github.com/MinaProtocol/mina/blob/2884275e6d4e18be6fd2d3afb19ae465d194e268/src/app/archive/create_schema.sql
https://github.com/MinaProtocol/mina/blob/2884275e6d4e18be6fd2d3afb19ae465d194e268/src/app/archive/zkapp_tables.sql
Note: zkApps Test Transaction CLI tool cannot be used to create token accounts in this release.

Complete Changelog since 05c2f73 in:
Github Compare: https://github.com/MinaProtocol/mina/compare/05c2f73d0f6e4f1341286843814ce02dcb3919e0...2884275e6d4e18be6fd2d3afb19ae465d194e268

Docker Images:
Every docker image is built for each of the supported platforms, replace CODENAME in the images below with your choice of bionic, focal, stretch, buster, or bullseye to select the base docker image.
Mina Daemon Docker Image: gcr.io/o1labs-192920/mina-daemon:1.3.2beta2-rampup-2884275-focal-berkeley
Mina Archive Docker Image: gcr.io/o1labs-192920/mina-archive:1.3.2beta2-rampup-2884275-focal

Step by Step Guide:
This version will only connect and function properly on the newly launched Berkeley Testnet.

Create a libp2p keypair for a node the first time and persist it
mina libp2p generate-keypair -privkey-path <path-to-the-key-file>

Run mina daemon

Set environment variable MINA_LIBP2P_PASS with the password set for the libp2p keypair in step1
Make sure to run the daemon with -libp2p-keypair <path-to-the-key-file> and --peer-list-url https://storage.googleapis.com/seed-lists/berkeley_seeds.txt.
If you are running the correct version on the correct network BERKELEY TESTNET, mina client status will show:

Chain id: 8fab56ef04619087270631e89bde75990d7968309f4008e3cac42eee9601fc96
Git SHA-1:  2884275e6d4e18be6fd2d3afb19ae465d194e268
```

**Comment by stevenplatt** [comment]
This item has been completed sucessfully.

**Comment by shimkiv** [comment]
@stevenplatt Can you please release native packages like [this](https://s3.console.aws.amazon.com/s3/object/packages.o1test.net?region=us-west-2&prefix=pool/focal/m/mi/mina-berkeley_1.3.2beta2-rampup-2884275.deb) and publish GH Discussion like [this](https://github.com/MinaProtocol/mina/discussions/12217).

**Comment by trevorbernard** [comment]
Also, is anybody else having an issue with the `berkeley_seeds.txt`?

We keep getting the following error:
```
...
2023-02-28 02:14:01 UTC [Error] libp2p_helper: failed to routing advertise: failed to find any peer in table
2023-02-28 02:14:01 UTC [Warn] starting libp2p up failed: $error
        error: {
  "commit_id": "[DIRTY]2884275e6d4e18be6fd2d3afb19ae465d194e268",
  "string": "libp2p error: failed to find any peer in table"
}
2023-02-28 02:14:46 UTC [Error] Failed to find any peers during initialization
```

**Comment by Isaac-DeFrain** [comment]
For more context
```
2023-02-28 16:28:03 UTC [Error] libp2p_helper: failed to set log level debug for mplex: error: No such logger
2023-02-28 16:28:03 UTC [Error] libp2p_helper: failed to set log level info for addrutil: error: No such logger
2023-02-28 16:28:03 UTC [Error] libp2p_helper: failed to set log level debug for stream-upgrader: error: No such logger
2023-02-28 16:28:03 UTC [Error] libp2p_helper: failed to set log level debug for discovery: error: No such logger
2023-02-28 16:28:03 UTC [Info] Network not instantiated when node status requested
2023-02-28 16:28:03 UTC [Info] Starting transition router
2023-02-28 16:28:03 UTC [Info] Initializing transition router
2023-02-28 16:28:03 UTC [Warn] GCLOUD_KEYFILE environment variable not set. Must be set to use upload_blocks_to_gcloud
2023-02-28 16:28:03 UTC [Info] Mina daemon is connecting
2023-02-28 16:28:03 UTC [Info] Initializing plugins
2023-02-28 16:28:03 UTC [Info] Not running uptime service, no URL given
2023-02-28 16:28:03 UTC [Info] Stopping daemon after 10440 mins and when there are no blocks to be produced
2023-02-28 16:28:03 UTC [Info] Attempted to turn on snark worker, but snark worker key is set to none
2023-02-28 16:28:03 UTC [Info] Daemon ready. Clients can now connect
2023-02-28 16:28:03 UTC [Info] Created GraphQL server at: http://localhost:3085/graphql
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to connect to peer: {12D3KooWAdgYL6hv18M3iDBdaK1dRygPivSfAfBNDzie6YqydVbs: [/dns4/seed-1.berkeley.o1test.net/tcp/10000]}failed to dial 12D3KooWAdgYL6hv18M3iDBdaK1dRygPivSfAfBNDzie6YqydVbs:
  * [/ip4/34.71.96.145/tcp/10000] failed to negotiate security protocol: read tcp4 <My_IP>:8302->34.71.96.145:10000: read: connection reset by peer
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to connect to peer: {12D3KooWLjs54xHzVmMmGYb7W5RVibqbwD1co7M2ZMfPgPm7iAag: [/dns4/seed-2.berkeley.o1test.net/tcp/10001]}failed to dial 12D3KooWLjs54xHzVmMmGYb7W5RVibqbwD1co7M2ZMfPgPm7iAag:
  * [/ip4/34.122.102.211/tcp/10001] dial backoff
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to connect to peer: {12D3KooWEiGVAFC7curXWXiGZyMWnZK9h8BKr88U8D5PKV3dXciv: [/dns4/seed-3.berkeley.o1test.net/tcp/10002]}failed to dial 12D3KooWEiGVAFC7curXWXiGZyMWnZK9h8BKr88U8D5PKV3dXciv:
  * [/ip4/34.70.146.0/tcp/10002] dial backoff
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to connect to peer: {12D3KooWAdgYL6hv18M3iDBdaK1dRygPivSfAfBNDzie6YqydVbs: [/dns4/seed-1.berkeley.o1test.net/tcp/10000]}failed to dial 12D3KooWAdgYL6hv18M3iDBdaK1dRygPivSfAfBNDzie6YqydVbs:
  * [/ip4/34.71.96.145/tcp/10000] dial backoff
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to connect to peer: {12D3KooWLjs54xHzVmMmGYb7W5RVibqbwD1co7M2ZMfPgPm7iAag: [/dns4/seed-2.berkeley.o1test.net/tcp/10001]}failed to dial 12D3KooWLjs54xHzVmMmGYb7W5RVibqbwD1co7M2ZMfPgPm7iAag:
  * [/ip4/34.122.102.211/tcp/10001] dial backoff
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to connect to peer: {12D3KooWEiGVAFC7curXWXiGZyMWnZK9h8BKr88U8D5PKV3dXciv: [/dns4/seed-3.berkeley.o1test.net/tcp/10002]}failed to dial 12D3KooWEiGVAFC7curXWXiGZyMWnZK9h8BKr88U8D5PKV3dXciv:
  * [/ip4/34.70.146.0/tcp/10002] dial backoff
2023-02-28 16:28:18 UTC [Error] libp2p_helper: failed to routing advertise: failed to find any peer in table
2023-02-28 16:28:18 UTC [Warn] starting libp2p up failed: $error
        error: {
  "commit_id": "[DIRTY]2884275e6d4e18be6fd2d3afb19ae465d194e268",
  "string": "libp2p error: failed to find any peer in table"
}
2023-02-28 16:29:03 UTC [Error] Failed to find any peers during initialization (crashing because this is not a seed node)
```

**Comment by stevenplatt** [comment]
@trevorbernard and @Isaac-DeFrain - thanks for reporting the errors and thanks for your patience. This behavior was validated and a fix was applied and packaged with new container release. Using the new container should resolve the behavior you are seeing. All the testnet nodes have also been updated to run the new images.

The new container name is:
- gcr.io/o1labs-192920/mina-daemon:2.0.0rampup1-fix-bootstrap-failure-rampup-d105e9f-buster-berkeley

For archive services, the conatiner is:
- gcr.io/o1labs-192920/mina-archive:2.0.0rampup1-rampup-ae78a64-bullseye

Release notes for the new container are now posted as a GH discussion available here: 
- https://github.com/MinaProtocol/mina/discussions/12757

**Comment by stevenplatt** [comment]
Closing this item to keep it isolated to the scope of deploying the testnet. @lk86 is currently working on .deb package releases for berkeley / rampup github branches.
